### PR TITLE
RsFileSink writes buffers to a file

### PR DESCRIPTION
**Pull request by luisbg**

### CODE REVIEW

File: src/rsfilesink.rs [hunk 1]
```diff
@@ -76,7 +76,16 @@ impl Sink for FileSink {
...
             None => return false,
             Some(ref location) => {
-                return true;
+                match File::create(location.as_path()) {
+                    Ok(file) => {
+                        self.file = Some(file);
+                        return true;
+                    },
+                    Err(err) => {
+                        println_err!("Could not open file for writing '{}': {}", location.to_str().unwrap_or("Non-UTF8 path"), err.to_string());
```
Comment (sdroege) [inline src/rsfilesink.rs]: Why to_str()? location already is a String so you can just print it with {} :)

File: src/rsfilesink.rs [hunk 2]
```diff
@@ -88,8 +97,20 @@ impl Sink for FileSink {
         true
     }
 
-    fn render(&mut self) -> Result<usize, GstFlowReturn> {
-        println!("Got a buffer!");
-        Err(GstFlowReturn::Ok)
+    fn render(&mut self, data: &mut [u8]) -> Result<(), GstFlowReturn> {
```
Comment (sdroege) [inline src/rsfilesink.rs]: This should just return a GstFlowReturn

File: src/rssink.rs [hunk 3]
```diff
@@ -37,11 +38,13 @@ pub extern "C" fn sink_get_uri(ptr: *mut Box<Sink>) -> *mut c_char {
 }
 
 #[no_mangle]
-pub extern "C" fn sink_render(ptr: *mut Box<Sink>) -> GstFlowReturn {
+pub extern "C" fn sink_render(ptr: *mut Box<Sink>, data_ptr: *mut u8, data_len_ptr: *mut usize) -> GstFlowReturn {
```
Comment (sdroege) [inline src/rssink.rs]: You can just make that ..., data_len: usize)

**Comment by luisbg** [comment]
Updated. Thanks for the review :)

**Comment by sdroege** [comment]
Merged

**Pull request closed without merging**